### PR TITLE
Build: macOS fixes (x64) and non-root system setup

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -263,6 +263,8 @@ ifneq ($(findstring centos,$(OSNICK)),)
 VECSIM_MARCH ?= skylake-avx512
 else ifneq ($(findstring xenial,$(OSNICK)),)
 VECSIM_MARCH ?= skylake-avx512
+else ifneq ($(findstring macos,$(OS)),)
+VECSIM_MARCH ?= skylake-avx512
```
Comment (chayim) [inline Makefile:267]: Curious - what is this

**Review by chayim — DISMISSED** [review]
Ah - python2 - I see it in numpy.

File: Makefile (hunk 1 above)
Comment (rafie) [reply Makefile:267]: It's for building with the best instruction set available (vecsim can support multiple and select the best one at runtime, but gcc does not always have the one you really want on a given platform - at least until we start building our own gcc).

**Review by oshadmi — APPROVED** [review]
Cool!

**Pull request merged**